lab-doug/README.md

```diff
@@ -0,0 +1,23 @@
+#Building a single-resource-api Router  
+
+#What I learned in lab-07
+*promises: getting comfortable with the basic syntax and the order of execution  
+*how to tie together the server, router object, route object, and registering the routes  
+*the value of abstracting simple repetitive tasks like err responses  
+*the value of adding additional properties to a req or res object so that the new property can be passed along and accessed by other methods  
+*getting the httpie commands right  
+*reaquainted myself with packet sniffers after a 10 year break.  By using   wireshark packet sniffer I could quickly tell if my http requests sent by httpie   were correctly formatted.  
+*learned the basics of Postman plug-in for chrome  
+*I learned the purpose of done() as it relates to mocha and where to place the call to the function when you have a series of nested requests  
+*learn some basics of testing with chai.  
+
+****
+#What I learned in Lab-08
+*For the first time, I could actually visualize the program before I started coding and determine what code needed to be modified to change the service to meet the objectives of lab-08.  Although I did not know the exact code that needed to be written, I understood it in general terms before I started coding.
+
+****
+#What items do I still not understand or what gaps have I identified in my knowledge?
+*I was really strong at using the browser debugging tool.  I am very weak at using debugging tools with node.  
+*how to automate test when you have dynamically generated items like UUID?  We did a work around in GET test that required nested requests.  Not sure if that is a good solution.  
+*how do you list multiple files under your 'test' task in mocha-gulp?  
+*need more work with mocha gulp-mocha and chai.  Tests are not running properly from gulp file
```

Comment:
One of the best ways to learn is to take the time to refactor code and try to make it better yourself. If you keep doing reflections like these (which is a great idea!) I would add a section for  what you would do differently next time.